### ISSUE TITLE
Improve URL input and Image URL input user interfaces consistency

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58507]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/pull/58505#issuecomment-1919314166

Consistency in user interfaces is part of accessibility, better user experience and consistent design.
A consistent UI reduces cognitive load, provides predictability and improves trust and affordance.

To insert links on text and images there are two different components. Besides obvious specific differences, for example images do have specific features like 'preset' to add a link to the image file or attachment, ideally these two UIs should look and behave very similarly. There's some pretty evidend inconsistencies though. To mention a few:

- The buttons placement: visually placed inside or outside the input field.
- The aria-disabled state of the Submit button.
- The labelling of the button: Apply vs. Submit.
- Different placeholder text: `Paste URL or type to search` vs. `Search or type url` with `url` lowercase (sigh).
- Divider border, spacing, and other CSS details.
- Possibly more. 

Additionally, while working on https://github.com/WordPress/gutenberg/pull/58505 I observed the state of the CSS isn't ideal. There's some overriding of default styles that should be avoided and, generally, working on this part of the CSS isn't a great user experience. It's likely that, by reducing functional inconsistencies and inconsistent markup, it is then posisble to freatly simplify the CSS as well.

Current state of the Image URL input UI before https://github.com/WordPress/gutenberg/pull/58505

![before 01](https://github.com/WordPress/gutenberg/assets/1682452/7814835a-5bfb-42e8-8ab0-8298b3dd5b94)

Current state of the 'default' URL input UI:

![default 01](https://github.com/WordPress/gutenberg/assets/1682452/0af0344e-e0d8-45fc-a625-2d2c193bf94b)

### Step-by-step reproduction instructions

- Select an Image block and add a link to ig.
- The popover with the URL input UI opens.
- Compare this UI with the one to add links to normal text.
- COmpare both visuals, interaction, and functionality.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Noting that the image link _settings_ spacing, left/right padding and top border look broken on trunk. Screenshot from current trunk.

In https://github.com/WordPress/gutenberg/pull/58505 I'm going to remove the top border. Further adjustments should be done in the context of this issue.

![Screenshot 2024-02-01 at 14 24 54](https://github.com/WordPress/gutenberg/assets/1682452/fd0a49e6-98b0-41fd-9373-48a41374f170)

**Comment by afercia** [comment]
The link _settings_ are largely inconsistent as well. For example:
- Checkboxes vs. toggle.
- Link `rel` vs. `no follow`.
- Link CSS class only in one of the two interfaces.
- Vertical spacing between input fields, left / right padding, borders.
- For the default URL input UI, the settings are available only _after_ a link has been applied, while for the image they are available also _before_.
- Cancel / Save buttons (related to the above point and the overall insert / save flow).

![link settings](https://github.com/WordPress/gutenberg/assets/1682452/a11d0874-5cee-460e-88f9-5d249924f1bd)

**Comment by im3dabasia** [comment]
Hey @afercia ,


## URLInput

URLInput is used in the following places:

1. LinkControl
2. URLPopover
3. SocialLink
4. URLPopoverButton

### LinkControl 

Have a detailed summary down below

### URLPopover

Placeholder: Paste or type URL
Button Label: Apply

<img width="652" alt="Image" src="https://github.com/user-attachments/assets/d1479fe0-54eb-41bc-8ca6-e9bf09904788" />


### SocialLink 

Placeholder: Enter social link
Button Label: Apply

<img width="398" alt="Image" src="https://github.com/user-attachments/assets/efa85ca8-eece-4eb8-8edc-aadf89685c4e" />

### URLInputButton 

Is not used in the codebase barring the test/ folders


## LinkControl

LinkControl is used in the following places:

1. MediaReplaceFlow
2. Button
3. NavigationLink
4. FormatLibrary

### MediaReplaceFlow 

Placeholder: Search or type url
Button Label: Used custom buttons via suffix


<img width="701" alt="Image" src="https://github.com/user-attachments/assets/872722e8-e5d4-48e7-8753-cd46adab1081" />


Since the suggestions are false for this particular case, I think we should consider changing the placeholder to simply 'Paste or type URL', similar to how it is done in Insert Media URL (the first photo). This would bring consistency.

<details>

<summary> Code for MediaReplaceFlow with suggestions false  </summary>

https://github.com/WordPress/gutenberg/blob/8b88ada2fd675509bf0c39a55c23a75cc67987cb/packages/block-editor/src/components/media-replace-flow/index.js#L223-L236

</details>

### NavigationLink 

Placeholder: Search or type URL
Button label: Submit

<img width="390" alt="Image" src="https://github.com/user-attachments/assets/e005e7fb-6548-4cdb-95cc-afddebb1cadc" />

### Button

Placeholder: Search or type URL
Button label: Submit

<img width="646" alt="Image" src="https://github.com/user-attachments/assets/cba8b13e-53dd-45b4-a9c1-987a3d0d920f" />

### FormatLibrary

Placeholder: Search or type URL
Button label: Submit

<img width="404" alt="Image" src="https://github.com/user-attachments/assets/ce53ea60-f3c5-434a-af64-0e967f16a4e8" />


I think we can improve on the following:

1. Work on the placeholders

- Currently the MediaReplaceFlow is wrong, i.e., it does not have the search functionality, but we have it mentioned in the placeholder.
- In the SocialLink, we have used the placeholder "Enter social link"; these wordings are completely different, i.e., link/url, Enter/type.


2. Decide on "Apply" vs "Submit" button label.



@afercia Please let me know what you think on this above mentioned points. looking forward.
